src/types.ts

```diff
@@ -213,3 +234,54 @@ export type FailureFunctionPayload = {
  * Makes all fields except the ones selected required
  */
 export type RequiredExceptFields<T, K extends keyof T> = Omit<Required<T>, K> & Partial<Pick<T, K>>;
+
+export type Waiter = {
+  url: string;
+  deadline: number;
+  headers: Record<string, string[]>;
+  timeoutUrl: string;
+  timeoutBody: unknown;
+  timeoutHeaders: Record<string, string[]>;
```

Comment:
Are these unused?

Reply:
I don't know what they are used for but they were undefined when I was testing.

Reply:
they are used, you should receive them if you call getWaiters. I am updating the integration test to check these fields too